pyomo/contrib/doe/doe.py

```diff
@@ -494,8 +496,11 @@ def _direct_kaug(self):
         for par in self.param.keys():
             cuid = pyo.ComponentUID(par)
             var = cuid.find_component_on(mod)
-            var.setlb(self.param[par])
-            var.setub(self.param[par])
+
+            # only set up bounds if they are variables
+            if isinstance(var, (pcb.var.Var, pcb.var.IndexedVar, pcb.var.ScalarVar)):
+                var.setlb(self.param[par])
+                var.setub(self.param[par])
```

Comment:
Why are you setting the bounds to be equal instead of just fixing the variable?

Reply:
I change these sentences to fixing the variable.